File: app/src/main/java/com/example/edt73/BodyMass.java
package com.example.edt73;

public class BodyMass {


    private GENDER gender;
    private MORPHOLOGY morphology;

    //formulas

    public static double FatMass(int altura, int circunferencia, GENDER gender){
        if(gender.equals(GENDER.MALE)){
            double result=Constant.RelativeHombre-(20*altura)/circunferencia;
            return Math.ceil(result * 100.0) / 100.0;
        }
        else{
            double result=Constant.RelativeMuejer-(20*altura)/circunferencia;
            return Math.ceil(result * 100.0) / 100.0;
        }
    }


    public static double WanderVael(int altura, GENDER gender){
        if(gender.equals(GENDER.MALE)){
            double result=(altura-150)*Constant.WanderHombres+50;
            return Math.round(result * 100.0) / 100.0;
        }
        else{
            double result=(altura-150)*Constant.WanderMujeres+50;
            return Math.round(result * 100.0) / 100.0;
        }
    }

    public static double Lorentz(int altura,int edad, GENDER gender){
        if(gender.equals(GENDER.MALE)){
            double result=altura-100-(double)(altura-150)/4+(double)(edad-20)/Constant.LorentzHombres;

            return Math.round(result * 100.0) / 100.0;
        }
        else{
            double result=altura-100-(double)(altura-150)/4+(double)(edad-20)/Constant.LorentzMujeres;
            return Math.round(result * 100.0) / 100.0;
        }
    }

    public static double Creff(int altura,int edad, MORPHOLOGY morphology){
        if(morphology.equals(MORPHOLOGY.SMALL)){
            double result=(double)(altura-100+edad/10)*Math.pow(0.9,2);
            return Math.round(result * 100.0) / 100.0;
        }
        else if(morphology.equals(MORPHOLOGY.MEDIUM)){
            double result=(double)(altura-100+edad/10)*0.9;
            return Math.round(result * 100.0) / 100.0;
        }
        else{
            double result=(double)(altura-100+edad/10)*0.9*1.1;
            return Math.round(result * 100.0) / 100.0;
        }
    }




}

File: app/src/main/java/com/example/edt73kotlin/BodyMass.kt
package com.example.edt73kotlin

class BodyMass {

        //formulas
        fun FatMass(altura: Int, circunferencia: Int, gender: GENDER?): Double {
            //si puede ser null has de poner name:DATO?
            //aunque le hayas puesto la condicion que no puede serlo, siempre enviara ?
            //por lo que has de cambiar lo que recibe

            return if (gender==GENDER.MALE) {
                val result = (Constant.RelativeHombre - 20 * altura / circunferencia).toDouble()
                Math.round(result * 100.0) / 100.0
            }
            else if(gender==GENDER.FEMALE) {
                val result = (Constant.RelativeMuejer - 20 * altura / circunferencia).toDouble()
                Math.round(result * 100.0) / 100.0
            }
            else{return -1.0;}
        }

        fun WanderVael(altura: Int, gender: GENDER?): Double {
            return if (gender==GENDER.MALE) {
                val result = (altura - 150) * Constant.WanderHombres + 50
                Math.round(result * 100.0) / 100.0
            }
            else if(gender==GENDER.FEMALE){
                val result = (altura - 150) * Constant.WanderMujeres + 50
                Math.round(result * 100.0) / 100.0
            }
            else{return -1.0;}
        }

        fun Lorentz(altura: Int, edad: Int, gender: GENDER?): Double {

            return if (gender==GENDER.MALE) {
                val result =
                    altura - 100 - (altura - 150).toDouble() / 4 + (edad - 20).toDouble() / Constant.LorentzHombres
                Math.round(result * 100.0) / 100.0
            }
            else if(gender==GENDER.FEMALE){
                val result =
                    altura - 100 - (altura - 150).toDouble() / 4 + (edad - 20).toDouble() / Constant.LorentzMujeres
                Math.round(result * 100.0) / 100.0
            }
            else{return -1.0;}
        }

        fun Creff(altura: Int, edad: Int, morphology: MORPHOLOGY?): Double {
            return if (morphology == MORPHOLOGY.SMALL) {
                val result = (altura - 100 + edad / 10).toDouble() * Math.pow(0.9, 2.0)
                Math.round(result * 100.0) / 100.0
            }
            else if (morphology == MORPHOLOGY.MEDIUM) {
                val result = (altura - 100 + edad / 10).toDouble() * 0.9
                Math.round(result * 100.0) / 100.0
            }
            else if(morphology == MORPHOLOGY.BROAD) {
                val result = (altura - 100 + edad / 10).toDouble() * 0.9 * 1.1
                Math.round(result * 100.0) / 100.0
            }
            else{return -1.0;}
        }

}
